### PR TITLE
feat: add KYCViewer migration

### PR DESCRIPTION
**Pull request by fedealconada**
- adds migration to deploy a new KYCViewer proxy and the implementation

### CODE REVIEW

File: script/migrations/utils/MigrationHelper.sol [hunk 1]
```diff
@@ -19,42 +20,58 @@ import "forge-std/Script.sol";
...
+    /// @dev deploys proxy contract via factory from deployer address
+    function _deployProxy(string memory contractName, address implementation) internal returns (address _proxy) {
         bool isEntryPoint = keccak256(abi.encodePacked(contractName)) == keccak256(abi.encodePacked("EntryPoint"));
         bool isWallet = keccak256(abi.encodePacked(contractName)) == keccak256(abi.encodePacked("KintoWallet"));
-        _proxy = _getChainDeployment(contractName);
+        // _proxy = _getChainDeployment(contractName);
 
         if (isWallet || isEntryPoint) revert("EntryPoint and KintoWallet do not use UUPPS Proxy");
-        if (_proxy != address(0)) revert("Proxy contract already deployed");
+        // if (_proxy != address(0)) revert("Proxy contract already deployed");
```
Comment (rrecuero) [inline script/migrations/utils/MigrationHelper.sol:47]: uncomment or remove

File: script/migrations/utils/MigrationHelper.sol [hunk 2]
```diff
@@ -19,42 +20,58 @@ import "forge-std/Script.sol";
...
     }
 
-    /// @dev deploys proxy contract from deployer
-    function deployProxy(string memory contractName, address implementation) public returns (address _proxy) {
+    /// @dev deploys proxy contract via factory from deployer address
+    function _deployProxy(string memory contractName, address implementation) internal returns (address _proxy) {
         bool isEntryPoint = keccak256(abi.encodePacked(contractName)) == keccak256(abi.encodePacked("EntryPoint"));
         bool isWallet = keccak256(abi.encodePacked(contractName)) == keccak256(abi.encodePacked("KintoWallet"));
-        _proxy = _getChainDeployment(contractName);
+        // _proxy = _getChainDeployment(contractName);
```
Comment (rrecuero) [inline script/migrations/utils/MigrationHelper.sol:44]: uncomment or remove

File: script/migrations/utils/MigrationHelper.sol [hunk 3]
```diff
@@ -103,63 +120,84 @@ contract MigrationHelper is Create2Helper, ArtifactsReader, UserOp {
...
+    // TODO: should be extended to work with other initalize() that receive params
     function _initialize(address _proxy, uint256 _signerPk) internal {
-        address payable _from = payable(_getChainDeployment("KintoWallet-admin"));
-
         // fund _proxy in the paymaster
-        require(
-            ISponsorPaymaster(payable(_getChainDeployment("SponsorPaymaster"))).balances(_proxy) > 0,
-            "Need to fund proxy in paymaster"
-        );
+        if (ISponsorPaymaster(payable(_getChainDeployment("SponsorPaymaster"))).balances(_proxy) == 0) {
```
Comment (rrecuero) [inline script/migrations/utils/MigrationHelper.sol:126]: This is needed only if the contract is not whitelisted by geth righ?

Comment (fedealconada) [reply script/migrations/utils/MigrationHelper.sol:126]: good call

**Review by rrecuero — APPROVED** [review]

**Pull request merged**